controllers/user.server.controller.js

```diff
@@ -93,72 +136,80 @@ exports.userViewById = function(req, res) {
 
 /**
  * Modifies the data of an existing forum user matching a given ID using HTTP request object data
- * @param req HTTP request object containing the user document fields being updated
- * @param res HTTP request response object containing the updated user document data
+ * @param req HTTP request object containing the user fields being updated, user ID and auth token for verification
+ * @param res HTTP request response object status code and updated user data in JSON format or error message
  */
 exports.userUpdateById = function(req, res) {
     const
         reqParams = req.params,
+        authToken = req.get(configParams.get('authToken')),
         reqBody = req.body;
-    let userUpdateParams,
-        isBadRequest = !isValidDocumentID(reqParams.id); // Check that a valid ID exists in the request parameters
 
-    if (!isBadRequest) {
-        // Set fields for updating to an object with either passed values or false to declare them as invalid
-        userUpdateParams = {
-            "username": reqBody.username && reqBody.username.length > 2 ? reqBody.username : false,
-            "displayName": reqBody.displayName && reqBody.displayName.length > 2 ? reqBody.displayName : false,
-            "email": reqBody.email && emailValidator.validate(reqBody.email) ? reqBody.email : false,
-            "hashedPassword": reqBody.password && reqBody.password.length > 0 ? reqBody.password : false
-        }
-        isBadRequest = !isAnyFieldValid(userUpdateParams); // Checks for any valid field, removes all invalid fields
+    // Set fields for updating to an object with either passed values or false to declare them as invalid
+    let userUpdateParams = {
+        "username": reqBody.username && reqBody.username.length > 2 ? reqBody.username : false,
+        "displayName": reqBody.displayName && reqBody.displayName.length > 2 ? reqBody.displayName : false,
+        "email": reqBody.email && emailValidator.validate(reqBody.email) ? reqBody.email : false,
+        "hashedPassword": reqBody.password && reqBody.password.length > 0 ? reqBody.password : false
     }
 
-    if (!isBadRequest) {
-        const isUserAuthenticated = true; // TODO: implement user authentication
-
-        if (isUserAuthenticated) {
-            User.updateUserById(reqParams.id, userUpdateParams, function (result) {
+    if (isValidDocumentID(reqParams.id) && isAnyFieldValid(userUpdateParams)) {
+        User.isUserAuthorized(reqParams.id, authToken, function(result) {
+            if (result.isAuth) {
+                User.updateUserById(reqParams.id, userUpdateParams, function (result) {
+                    if (result.err) {
+                        // Return the error message with the error status
+                        res.status(result.status).send(result.err);
+                    } else {
+                        // Return the forum post document object with 201 status
+                        res.status(201).json({"updatedForumUser": result});
```

Comment:
```suggestion
                        res.status(201).json({ "updatedForumUser": result });
```

Reply:
I don't think there is any specific style we have to follow regading this